plugins/woocommerce/tests/e2e-pw/tests/merchant/create-page.spec.js

```diff
@@ -36,38 +37,38 @@ test.describe( 'Can create a new page', () => {
 
 		const welcomeModalVisible = await page
 			.getByRole( 'heading', {
-				name: 'Welcome to the block editor',
+				name: `${getTranslationFor('Welcome to the block editor')}`,
 			} )
 			.isVisible();
 
 		if ( welcomeModalVisible ) {
-			await page.getByRole( 'button', { name: 'Close' } ).click();
+			await page.getByRole( 'button', { name: getTranslationFor('Close') } ).click();
 		}
 
 		await page
-			.getByRole( 'textbox', { name: 'Add Title' } )
+			.getByRole( 'textbox', { name: getTranslationFor('Add Title') } )
 			.fill( pageTitle );
 
-		await page.getByRole( 'button', { name: 'Add default block' } ).click();
+		await page.getByRole( 'button', { name: getTranslationFor('Add default block') } ).click();
 
 		await page
 			.getByRole( 'document', {
 				name:
-					'Empty block; start writing or type forward slash to choose a block',
+				`${getTranslationFor('Empty block; start writing or type forward slash to choose a block')}`,
 			} )
 			.fill( 'Test Page' );
 
 		await page
-			.getByRole( 'button', { name: 'Publish', exact: true } )
+			.getByRole( 'button', { name: getTranslationFor('Publish'), exact: true } )
 			.click();
 
 		await page
-			.getByRole( 'region', { name: 'Editor publish' } )
-			.getByRole( 'button', { name: 'Publish', exact: true } )
+			.getByRole( 'region', { name: `${getTranslationFor('Editor publish')}` } )
+			.getByRole( 'button', { name: `${getTranslationFor('Publish')}`, exact: true } )
```

Comment:
I think this is a great suggestion and I think it is worth the effort

Reply:
I wonder if we can [create our own selectors](https://playwright.dev/docs/api/class-selectors) in Playwright to embed the `getForTranslation()` function inside new selectors.

For example, I think with this Playwright feature we could do:
`page.locator('translatedText=Editor Publish');`

This would be more readable and maintainable than:
```
const editorPublish = getTranslationFor( 'Editor publish' );
page.getByRole( 'region', { name: editorPublish } )
```

I'm aware that in this case, we're not using `text` but getByRole() but based on the Playwright documentation, I think this could cover this as well.

That said, this is an unknown feature to me so we might need to check if it can cover this use case. It's worth a quick test though as the code would be very clean and easy to use.